azkaban-common/src/test/resources/azkaban/executor/container/v1PodTestTemplate1.yaml

```diff
@@ -47,15 +55,13 @@ spec:
         initialDelaySeconds: 5
         periodSeconds: 5
       volumeMounts:
+        - mountPath: /export/some/path
```

Comment:
If I change the container name from `az-flow-container` in the template pod spec to `template-az-flow-container` for example, below spec portion still gets applied to container `az-flow-container` in the final pod spec
```
livenessProbe:
  exec:
    command:
      - cat
      - /tmp/liveProbe
  initialDelaySeconds: 5
  periodSeconds: 5
readinessProbe:
  exec:
    command:
      - cat
      - /tmp/readyProbe
```
Why is that? I would expect above yaml portion to be missing from final pod spec since container names didn't match. Or perhaps an error saying that the template pod spec can only contain 1 non init container with name `az-flow-container`.

Reply:
The template and the dymanic one which gets created by webserver are merged together if they have the same container names . For app/flow container, we need exactly one container, so if the names are different in the pod-spec and the one passed to the builder, this can occur. We need to change the name of "az-flow-container" at both places in pod-spec-template and in the builder test.